### ISSUE TITLE
[native-image] Catch TypeNotPresentException in enclosingMethodOrConstructor

### ISSUE THREAD

**Issue by tianxiaogu** [oracle/graal#1779]
```
Fatal error: java.lang.TypeNotPresentException: Type reactor.netty.DisposableServer not present
	at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:117)
	at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
	at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at sun.reflect.generics.repository.ConstructorRepository.getParameterTypes(ConstructorRepository.java:94)
	at java.lang.Class.getEnclosingMethod(Class.java:1026)
	at com.oracle.svm.reflect.hosted.ReflectionDataBuilder.enclosingMethodOrConstructor(ReflectionDataBuilder.java:332)
	at com.oracle.svm.reflect.hosted.ReflectionDataBuilder.duringAnalysis(ReflectionDataBuilder.java:206)
	at com.oracle.svm.reflect.hosted.ReflectionFeature.duringAnalysis(ReflectionFeature.java:82)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$8(NativeImageGenerator.java:714)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:63)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:714)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:447)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: reactor.netty.DisposableServer
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:448)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
	... 17 more
```

We got a `TypeNotPresentException` when some class is missing in the class path (even with `--allow-incomplete-classpath`)? 

Can we catch `TypeNotPresentException` in `com.oracle.svm.reflect.hosted.ReflectionDataBuilder#enclosingMethodOrConstructor` as well?

1. `TypeNotPresentException` is mentioned in <https://github.com/oracle/graal/issues/1438>,
2. The corresponding code is at <https://github.com/oracle/graal/blob/c25516d72d0b48635c68714cad89bed01e2935c2/substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/hosted/ReflectionDataBuilder.java#L309-L317>

**Comment by dzou** [comment]
Recently I encountered the same error.

**Comment by cstancu** [comment]
Fixed by 096d01ed8d8fe28b12ea0b54f199e30eea7b263d.
